### PR TITLE
Fix some Ageable Entities spawning as baby versions sometimes, and add more baby variants 

### PR DESCRIPTION
**Pull request by Maroon28**
Sometimes, adult entities used to spawn as babies. Not only that, but some of the baby variants were missing. This commit adds any missing variants and fixes the issue with adults turning to babies randomly, by making sure to set any

# Summary

* This is a…
    * [ X] Bug fix
    * [ X] Feature addition
* **Describe this change in 1-2 sentences**:
Sets any ageable entities to adults by default, and only sets them to babies when specified. Also includes missing baby types from #689 

# Problem

As described in issue #687, entities with ageable variants sometimes spawn as babies. And not only that, some of said baby variants are missing from the MACreature file, as described in issue #689. This pull request makes sure any entities that are not supposed to be babies are set as adults, and adds more baby types. 
* GitHub issue (_optional_): #687 and #689 

# Solution

Simply put, just turn any regular entities to adults, and only turn them into babies when specified. I would've made an if statement by checking `e instanceof ageable` then turning it into an adult if true, just to save a bit of branches. Sadly, that would break any old versions running Mobarena, since they don't have the ageable option.

### CODE REVIEW

File: src/main/java/com/garbagemule/MobArena/waves/MACreature.java [hunk 1]
```diff
@@ -241,6 +263,12 @@ private static void registerTypeVariants() {
         //
         put("angrybee", "angrybees", "BEE", null);
         put("angrywolf", "angrywolves", "WOLF");
+        put("babypiglin", "babypiglins", "PIGLIN");
+        put("babyzombifiedpiglin", "babyzombifiedpiglins", "ZOMBIFIED_PIGLIN");
+        put("babyhoglin", "babyhoglins", "HOGLIN");
+        put("babyzoglin", "babyzoglins", "ZOGLIN");
```
Comment (garbagemule) [inline src/main/java/com/garbagemule/MobArena/waves/MACreature.java:269]: These lines generate the following errors on 1.13:

```
[18:59:16 WARN]: [MobArena] Failed to register monster type 'babypiglin', because its type was not found: PIGLIN
[18:59:16 WARN]: [MobArena] Failed to register monster type 'babyzombifiedpiglin', because its type was not found: ZOMBIFIED_PIGLIN
[18:59:16 WARN]: [MobArena] Failed to register monster type 'babyhoglin', because its type was not found: HOGLIN
[18:59:16 WARN]: [MobArena] Failed to register monster type 'babyzoglin', because its type was not found: ZOGLIN
```

This is a result of the types piglin, hoglin, and zoglin not existing prior to 1.16, so they'll never resolve on earlier versions. It's the same issue as with angry bees, and the solution is to pass `null` as the last type. Check out the comment in the `put()` method that explains it in more detail.

The `babyzombifiedpiglin` is actually already present as a "broken" type further down (and constitutes a branch in the spawn method's switch statement), so it shouldn't need to be registered here as well.

Please re-sort the puts by the first entry so the list stays sorted. It helps with glanceability since the types are a bit all over the place version-wise :)

File: src/main/java/com/garbagemule/MobArena/waves/MACreature.java [hunk 2]
```diff
@@ -241,6 +263,12 @@ private static void registerTypeVariants() {
...
+        put("babyhusk", "babyhusks", "HUSK");
+        put("babydrowned", "babydrowneds", "DROWNED");
```
Comment (garbagemule) [inline src/main/java/com/garbagemule/MobArena/waves/MACreature.java:271]: Let's go with `babydowned` as both singular and plural, similar to `explodingsheep`.

File: src/main/java/com/garbagemule/MobArena/waves/MACreature.java [hunk 3]
```diff
@@ -131,29 +131,51 @@ public LivingEntity spawn(Arena arena, World world, Location loc) {
...
             case "babyzombievillager":
             case "babyzombie":
                 ((Zombie) e).setBaby(true);
                 break;
+            case "babyhoglin":
+                ((Hoglin) e).setImmuneToZombification(true);
+                ((Hoglin) e).setBaby();
+                break;
+            case "babypiglin":
+                ((PiglinAbstract)e).setBaby();
```
Comment (garbagemule) [inline src/main/java/com/garbagemule/MobArena/waves/MACreature.java:145]: Add a space between the first `)` and the `e` here for consistency.

**Review by garbagemule — CHANGES_REQUESTED** [review]
Looks good aside from the sub-1.16 warnings and a few minor details. I'll give it a real whirl some time this week so we can wrap this up :)

File: src/main/java/com/garbagemule/MobArena/waves/MACreature.java (hunk 2 above)
Comment (Nesseley) [reply src/main/java/com/garbagemule/MobArena/waves/MACreature.java:271]: Just pointing out the spelling mistake `downed` instead of `drowned` :)

Comment (Maroon28) [reply src/main/java/com/garbagemule/MobArena/waves/MACreature.java:271]: Yeah, wasn't sure which one to go with so I just added the s xD Will do

File: src/main/java/com/garbagemule/MobArena/waves/MACreature.java (hunk 1 above)
Comment (Maroon28) [reply src/main/java/com/garbagemule/MobArena/waves/MACreature.java:269]: Oh shoot. Will take care of that now

File: src/main/java/com/garbagemule/MobArena/waves/MACreature.java (hunk 3 above)
Comment (Maroon28) [reply src/main/java/com/garbagemule/MobArena/waves/MACreature.java:145]: Didn't see that one, yikes!

**Comment by Maroon28** [comment]
Just pushed a commit to make all these changes!
